daemon/container_operations.go

```diff
@@ -597,7 +597,10 @@ func cleanOperationalData(es *network.EndpointSettings) {
 }
 
 func (daemon *Daemon) updateNetworkConfig(container *container.Container, n libnetwork.Network, endpointConfig *networktypes.EndpointSettings, updateSettings bool) error {
-	if !containertypes.NetworkMode(n.Name()).IsUserDefined() {
+	endpointConfig.DNS = daemon.configStore.DNS
+
+	if !container.EnableServiceDiscoveryOnDefaultNetwork() &&
```

Comment:
I have moved this to the next line otherwise compose complains to bring up containers in the default nat network in windows.